starlark/value.go

```diff
@@ -1295,17 +1317,33 @@ func (s *Set) Intersection(other Iterator) (Value, error) {
 
 func (s *Set) SymmetricDifference(other Iterator) (Value, error) {
 	diff := s.clone()
+	if err := diff.symmetricDifferenceUpdate(s, other); err != nil {
+		return nil, err
+	}
+	return diff, nil
+}
+
+// s and sReadonlyCopy must contain the same elements; s will be modified,
```

Comment:
It's unfortunate that you need to clone s just to be able to iterate over it at the same time as (one of the callers) wishes to update s. Can `set_symmetric_difference_update` not simply call `s.symmetricDifferenceUpdate(s, iter)`? Is the awkwardness designed to handle the edge case `x.op_update(x)` for nonempty sets x? It might be simpler to shortcut in that case: x.symmetric_difference_update(x) becomes just x.clear().

We should add tests of such cases.

Reply:
The problem is that `other` can be a list/tuple containing duplicates. Consider something like `set([x]).symmetric_difference([x, x])` - if we didn't have sReadonlyCopy, we would naively iterate over `[x, x]`, delete `x`, re-insert `x`, and end up with `set([x])` as the wrong answer.

So we have two possibilities: either make a temporary copy of the lhs set `s`, as I did here; or else make a temporary set copy of the rhs list `other` to deduplicate.

With the second approach, it makes more sense to pass `other` as a `Value` rather than an iterator, so that we can avoid making a copy if `other` is already deduplicated (e.g. already a set). And that approach also can work for order-preserving intersection. WDYT?

Reply:
Ah, interesting example (and definitely worthy of a comment). Still I don't understand how the readonly copy of s solves that problem: we would need to first accumulate the iterator into a set s', and then compute sΔs', otherwise we would need to remember information (parity of occurrence) about all iterator values already seen.

> So we have two possibilities: either make a temporary copy of the lhs set s, as I did here; or else make a temporary set copy of the rhs list other to deduplicate.

The second approach seems conceptually simpler.

Reply:
Done.